src/test/java/htsjdk/samtools/CRAMFileWriterTest.java

```diff
@@ -210,64 +227,114 @@ public void testCRAMWriterNotPresorted() {
     public void testCRAMWriterConstructorRequiredReference_1() {
         final SAMFileHeader header = createSAMHeader(SAMFileHeader.SortOrder.coordinate);
         final ByteArrayOutputStream outStream = new ByteArrayOutputStream();
-        new CRAMFileWriter(outStream, null, header, null);
+        try (CRAMFileWriter ignored = new CRAMFileWriter(outStream, null, header, null)) {
+        }
     }
 
     @Test(description = "Test CRAMWriter constructor reference required 2", expectedExceptions = IllegalArgumentException.class)
     public void testCRAMWriterConstructorRequiredReference_2() {
         final SAMFileHeader header = createSAMHeader(SAMFileHeader.SortOrder.coordinate);
         final ByteArrayOutputStream outStream = new ByteArrayOutputStream();
-        new CRAMFileWriter(outStream, null, null, header, null);
+        try (CRAMFileWriter ignored = new CRAMFileWriter(outStream, null, null, header, null)) {
+        }
     }
 
     @Test(description = "Test CRAMWriter constructor reference required 3", expectedExceptions = IllegalArgumentException.class)
     public void testCRAMWriterConstructorRequiredReference_3() {
         final SAMFileHeader header = createSAMHeader(SAMFileHeader.SortOrder.coordinate);
         final ByteArrayOutputStream outStream = new ByteArrayOutputStream();
-        new CRAMFileWriter(outStream, null, true, null, header, null);
+        try (CRAMFileWriter ignored = new CRAMFileWriter(outStream, null, true, null, header, null)) {
+        }
     }
 
     @Test
     public void test_roundtrip_tlen_preserved() throws IOException {
-        SamReader reader = SamReaderFactory.make().open(new File("src/test/resources/htsjdk/samtools/cram_tlen_reads.sorted.sam"));
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        final ReferenceSource source = new ReferenceSource(new File("src/test/resources/htsjdk/samtools/cram_tlen.fasta"));
-        CRAMFileWriter writer = new CRAMFileWriter(baos, source, reader.getFileHeader(), "test.cram");
-        SAMRecordIterator iterator = reader.iterator();
-        List<SAMRecord> records = new ArrayList<SAMRecord>();
-        while (iterator.hasNext()) {
-            final SAMRecord record = iterator.next();
-            writer.addAlignment(record);
-            records.add(record);
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        final ReferenceSource source = new ReferenceSource(new File(SAM_TOOLS_TEST_DIR, "cram_tlen.fasta"));
+        final List<SAMRecord> records = new ArrayList<>();
+
+        try (SamReader reader = SamReaderFactory.make().open(new File(SAM_TOOLS_TEST_DIR, "cram_tlen_reads.sorted.sam"))) {
+            try (CRAMFileWriter writer = new CRAMFileWriter(baos, source, reader.getFileHeader(), "test.cram")) {
```

Comment:
You can merge these into 1 try statement with 2 resources which is probably cleaner